src/metadata_tool/tools/parsers.clj

```diff
@@ -21,35 +21,42 @@
             [hickory.select                       :as sel]
             [metadata-tool.sources.confluence     :as cfl]
             [metadata-tool.config                 :as cfg]
+            [clojure.pprint                       :as pp]
             [metadata-tool.sources.metadata       :as md]))
 
 (def not-nil? (complement nil?))
 
 (defn parse-string
   [s]
-  (-> s
-      (str/replace "&nbsp;" " ")
-      (str/replace "\u00A0" " ")
-      str/trim))
+  (if (empty? s)
+    ""
+    (-> s
+        (str/replace "&nbsp;" " ")
+        (str/replace "\u00A0" " ")
+        str/trim)))
+
+(defn resolve-acronym
+  [string]
+  (if-let [acronym (get (:acronyms (:meetings cfg/config)) string)]
+    acronym
+    string))
 
 (defn parse-name
   [string to-remove]
   (when-not (str/blank? string)
     (if (empty? to-remove)
       string
-      (if-let [acronym (get (:acronyms (:confluence cfg/config)) string)]
-        acronym
-        (parse-name
-         (str/replace string (first to-remove) "")
-         (rest to-remove))))))
+      (parse-name
```

Comment:
instead of recursion, you could built up a regex Pattern by cloure.string/joining the parts with `|`.  
Also, you won't need these guard clauses and nil/empty checks because it will just pass through.

Reply:
Thanks @tschady , I'll try to work on it; in the meantime, feel free to submit the code (or some pseudo-code) of the solution you have in mind.